modules/features/endofyear/src/main/java/au/com/shiftyjelly/pocketcasts/endofyear/StoriesFragment.kt

```diff
@@ -35,12 +46,32 @@ class StoriesFragment : BaseDialogFragment() {
                     StoriesScreen(
                         viewModel = viewModel,
                         onCloseClicked = { dismiss() },
+                        onShareClicked = { onCaptureBitmap ->
+                            viewModel.onShareClicked(
+                                onCaptureBitmap,
+                                requireContext(),
+                                ::showShareForFile
+                            )
+                        }
                     )
                 }
             }
         }
     }
 
+    private fun showShareForFile(file: File) {
+        val context = requireContext()
+        try {
+            val intent = Intent(Intent.ACTION_SEND)
+            intent.type = "image/png"
+            val uri = FileUtil.createUriWithReadPermissions(file, intent, requireContext())
+            intent.putExtra(Intent.EXTRA_STREAM, uri)
+            shareLauncher.launch(Intent.createChooser(intent, context.getString(LR.string.end_of_year_share_via)))
```

Comment:
This launches share intent with an edit button. On clicking "edit" from the chooser, an exception is seen in logcat:

```
Writing exception to parcel
java.lang.SecurityException: Permission Denial: writing androidx.core.content.FileProvider uri content://<path > requires the provider be exported, or grantUriPermission()
```

I tried to grant `FLAG_GRANT_WRITE_URI_PERMISSION` locally but it didn't help. Strangely, testing on an emulator shows logs even for reading the file. Also see related SO discussions [link 1](https://stackoverflow.com/questions/63723656/share-content-permission-denial-when-using-intent-createchooser), [link2](https://stackoverflow.com/questions/66521216/permission-denial-reading-androidx-core-content-fileprovider-trying-to-share-a).

Sharing and editing work without problems. I still have to see why these logs are shown. Let me know if you have any suggestions. Another option is to not show the edit button, the share dialog will look this way then:

<img width=320 src="https://user-images.githubusercontent.com/1405144/198999110-cc364f04-8159-4995-aacc-a0790c81e5d9.png"/>